File: Tests/ComputerBarTests/SSHConfigParserTests.swift
import XCTest
@testable import ComputerBar

final class SSHConfigParserTests: XCTestCase {
    func testParseAliasesIgnoresWildcardsAndNegatedPatterns() {
        let config = """
        Host *
            ServerAliveInterval 60

        Host dev staging
            User ubuntu

        Host !prod
            User nobody

        Host jump-* qa?
            Port 2222

        Host prod # inline comment
            HostName 10.0.0.1
        """

        XCTAssertEqual(SSHConfigParser.parseAliases(from: config), ["dev", "staging", "prod"])
    }

    func testParseResolvedHostUsesResolvedValues() {
        let output = """
        host dev
        hostname 192.168.1.30
        user ubuntu
        port 2222
        """

        let host = SSHConfigParser.parseResolvedHost(alias: "dev", output: output)

        XCTAssertEqual(host.alias, "dev")
        XCTAssertEqual(host.hostName, "192.168.1.30")
        XCTAssertEqual(host.user, "ubuntu")
        XCTAssertEqual(host.port, 2222)
    }
}

File: Windows/tests/ComputerBar.Core.Tests/SshConfigParserTests.cs
using ComputerBar.Core;

namespace ComputerBar.Core.Tests;

public sealed class SshConfigParserTests
{
    [Fact]
    public void ParseHostsIgnoresWildcardsAndNegatedPatterns()
    {
        const string config = """
        Host *
            ServerAliveInterval 60

        Host dev staging
            User ubuntu

        Host !prod
            User nobody

        Host jump-* qa?
            Port 2222

        Host prod # inline comment
            HostName 10.0.0.1
        """;

        var hosts = SshConfigParser.ParseHosts(config);

        Assert.Equal(["dev", "staging", "prod"], hosts.Select(host => host.Alias));
        Assert.Equal("ubuntu", hosts[0].User);
        Assert.Equal("10.0.0.1", hosts[2].HostName);
    }

    [Fact]
    public void ParseHostsUsesBlockValues()
    {
        const string config = """
        Host dev
          HostName 192.168.1.30
          User ubuntu
          Port 2222
        """;

        var host = Assert.Single(SshConfigParser.ParseHosts(config));

        Assert.Equal("dev", host.Alias);
        Assert.Equal("192.168.1.30", host.HostName);
        Assert.Equal("ubuntu", host.User);
        Assert.Equal(2222, host.Port);
    }
}
